tgui/packages/tgui/interfaces/AirAlarm.tsx

```diff
@@ -0,0 +1,480 @@
+  panicSiphonPath: string;
+  filteringPath: string;
+  modes: {
+    name: string;
+    desc: string;
+    path: string;
+    danger: BooleanLike;
+  }[];
+  thresholdTypeMap: Record<string, number>;
+};
+
+export const AirAlarm = (props, context) => {
+  const { act, data } = useBackend<AirAlarmData>(context);
+  const locked = data.locked && !data.siliconUser;
+  return (
+    <Window width={475} height={650}>
+      <Window.Content scrollable>
+        <InterfaceLockNoticeBox />
+        <AirAlarmStatus />
+        {!locked && <AirAlarmControl />}
+      </Window.Content>
+    </Window>
+  );
+};
+
+const AirAlarmStatus = (props, context) => {
+  const { data } = useBackend<AirAlarmData>(context);
+  const dangerMap = {
+    0: {
+      color: 'good',
+      localStatusText: 'Optimal',
+    },
+    1: {
+      color: 'average',
+      localStatusText: 'Caution',
+    },
+    2: {
+      color: 'bad',
+      localStatusText: 'Danger (Internals Required)',
+    },
+  };
+  const localStatus = dangerMap[data.dangerLevel] || dangerMap[0];
+  return (
+    <Section title="Air Status">
+      <LabeledList>
+        {(data.envData.length > 0 && (
+          <>
+            {data.envData.map((entry) => {
+              const status = dangerMap[entry.danger] || dangerMap[0];
+              return (
+                <LabeledList.Item
+                  key={entry.name}
+                  label={entry.name}
+                  color={status.color}>
+                  {entry.value}
+                </LabeledList.Item>
+              );
+            })}
+            <LabeledList.Item label="Local status" color={localStatus.color}>
+              {localStatus.localStatusText}
+            </LabeledList.Item>
+            <LabeledList.Item
+              label="Area status"
+              color={data.atmosAlarm || data.fireAlarm ? 'bad' : 'good'}>
+              {(data.atmosAlarm && 'Atmosphere Alarm') ||
+                (data.fireAlarm && 'Fire Alarm') ||
+                'Nominal'}
+            </LabeledList.Item>
+          </>
+        )) || (
+          <LabeledList.Item label="Warning" color="bad">
+            Cannot obtain air sample for analysis.
+          </LabeledList.Item>
+        )}
+        {!!data.emagged && (
+          <LabeledList.Item label="Warning" color="bad">
+            Safety measures offline. Device may exhibit abnormal behavior.
+          </LabeledList.Item>
+        )}
+      </LabeledList>
+    </Section>
+  );
+};
+
+const AIR_ALARM_ROUTES = {
+  home: {
+    title: 'Air Controls',
+    component: () => AirAlarmControlHome,
+  },
+  vents: {
+    title: 'Vent Controls',
+    component: () => AirAlarmControlVents,
+  },
+  scrubbers: {
+    title: 'Scrubber Controls',
+    component: () => AirAlarmControlScrubbers,
+  },
+  modes: {
+    title: 'Operating Mode',
+    component: () => AirAlarmControlModes,
+  },
+  thresholds: {
+    title: 'Alarm Thresholds',
+    component: () => AirAlarmControlThresholds,
+  },
+};
```

Comment:
```suggestion
} as const;
```

For all and any global-ish constants.

Using `as const;` prevents it from being mutated, and it allows us to generate a type that we can use to constrain the available screens: `type Screen  = keyof typeof AIR_ALARM_ROUTES;`